.github/workflows/examples-autotools.yaml

```diff
@@ -74,7 +74,7 @@ jobs:
         runs-on: ubuntu-latest
 
         container:
-            image: connectedhomeip/chip-build-nrf-platform:0.4.1
+            image: connectedhomeip/chip-build-nrf-platform:0.4.2
```

Comment:
is this image pushed yet?

Reply:
Yes and no :)
Initially I asked Andrei to rebuild nrf image with "--no-cache" flag so that fresh sources are picked up. Later we agreed that a cleaner approach would be using specific commit-ids in Dockerfile and not having to bypass the docker cache, so I updated the definition as you have seen. The 0.4.2 image which is on dockerhub right now is the one built with "--no-cache", so it's not the final version yet, but is good enough to build this PR.

Reply:
We will have to merge them ... I originally needed 0.4.2 to have a JDK in the android-build, so I built one and Damian noticed that the nRF build was odd. We probably want both the nRF build updates and JDK commits in before we create another 'real/final 0.4.3'.

That being said, I have no plans to push any other 0.4.2 until then so this should be safe.